src/structs.rs

```diff
@@ -107,6 +107,63 @@ impl From<&AssetCode> for BaseField {
     }
 }
 
+/// Amount value type
+#[tagged_blob("AMOUNT_VALUE")]
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    PartialEq,
+    Default,
+    Hash,
+    Eq,
+    From,
+    Into,
```

Comment:
Does this just do `From<u128>` and `Into<u128>`? Might also be useful to have `From` and `Into` for `u64`, `u32`, etc.